Modules/Sources/Storage/CoreData/CoreDataIterativeMigrator.swift

```diff
@@ -59,6 +59,18 @@ final class CoreDataIterativeMigrator {
         // Find the current model used by the store.
         let sourceModel = try model(for: sourceMetadata)
 
+        // Check if we should nuke the DB entirely based on the oldest supported data model.
+        // We will attempt to either move the merchant to the latest model version, or perform an iterative migration
+        // The persistent store coordinator will automatically create a fresh store with the current model as needed.
+        if shouldDestroyPersistentStore(for: sourceModel) {
+            do {
+                try persistentStoreCoordinator.destroyPersistentStore(at: sourceStoreURL, ofType: storeType, options: nil)
+                DDLogInfo("[CoreDataIterativeMigrator] Database at \(sourceStoreURL) destroyed successfully.")
+            } catch {
+                DDLogError("[CoreDataIterativeMigrator] Direct migration failed. Error: \(error). Falling back to iterative migration.")
```

Comment:
super nit: maybe the error message can match the operation that failed?
```suggestion
                DDLogError("[CoreDataIterativeMigrator] Database destruction failed. Error: \(error). Falling back to iterative migration.")
```